vercel/next.js
# chore: narrows regexp to enable middleware source maps

### What's in there?

This is a follow up of #37428
The opt-in plugin for middleware sourcemap is using a regexp, which I relaxed to detect both `middleware.` and `src/middleware.`

However, I've relaxed it too much (it used to start with `^`), and this PR reverts it.

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

### How to test

New tests to cover this case can be ran with `pnpm testheadless --testPathPattern production/generate-middleware-source-maps`
Tests for the overall src/ folder in middleware are: `pnpm testheadless --testPathPattern middleware-src`

To try it live:
1. make a simple app with a middleware in `src/` folder, let's say in typescript
   ```typescript
   // src/middleware.ts
   import type { NextRequest } from 'next/server'
   import { NextResponse } from 'next/server'
   
   export default function middleware(req: NextRequest) {
      return NextResponse.next()
   }
   ```
1. Include a dummy page
   ```js
   // src/pages.index.jsx
   export default function Home() {
     return <h1>Hi there!</h1>
   }
   ```
1. Enables middleware sourcemaps
   ```js
   // next.config.js
   module.exports = { experimental: { middlewareSourceMaps: true }}
   ```
1. Build your application `pnpm next build`
1. You should find the source maps in `.next/server/src/middleware.js.map`
